pdf-converter/app/pdf_processor.py

```
@@ -0,0 +1,130 @@
+import os
+
+class PDFProcessor:
+    def __init__(self, config_manager):
+        self.config = config_manager
+        self.client = config_manager.get_openai_client()
+    
+    def encode_image(self, image_path: str) -> str:
+        with open(image_path, "rb") as image_file:
+            return base64.b64encode(image_file.read()).decode("utf-8")
+    
+    def process_pdf(self, uploaded_file, progress_callback=None) -> pd.DataFrame:
```

Pylint (3.3.7)

[refactor] Too many local variables (21/15)

(R0914)